### PR TITLE
allow ecdsa-sha2-nistp256 hostkeys,  add host_aliases attribute to sshkey resource, add support for PubkeyAuthentication

### PR DESCRIPTION
**Pull request by florianfa**

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -472,6 +478,7 @@
     it { should contain_file('sshd_config').with_content(/^AuthorizedKeysCommand \/path\/to\/command$/) }
     it { should contain_file('sshd_config').with_content(/^AuthorizedKeysCommandUser asdf$/) }
     it { should contain_file('sshd_config').with_content(/^HostbasedAuthentication no$/) }
+    it { should contain_file('sshd_config').with_content(/^PubkeyAuthentication yes$/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: This should be set to a non-default value; in our case, 'no'.

**Comment by ghoneycutt** [comment]
Thanks for the work!

This should have been three separate PR's because three distinct things are being added, though it is simple enough that I can merge this after the spec test is updated.

**Comment by florianfa** [comment]
I've changed PubkeyAuthentication to no. Next time I try to make different PR's.
Regards Florian

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (florianfa) [inline spec/classes/init_spec.rb]: If I change the PubkeyAuthentication to no the test would break.

Comment (florianfa) [inline spec/classes/init_spec.rb]: additionally added param :sshd_pubkeyauthentication  => 'no'

**Comment by ghoneycutt** [comment]
Thank you!

Released in v3.39.0

**Pull request merged**